internal/util/debug/archive_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debug
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"slices"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/FerretDB/FerretDB/internal/util/state"
+)
+
+func TestArchiveHandler(t *testing.T) {
+	t.Parallel()
+
+	// during test some of the files from the below list shall be absent
+	fileList := []string{
+		"metrics", "heap",
+	}
+
+	host := "127.0.0.1:5454"
```

Comment:
That will fail if the port is occupied by something else.

Listening on `0.0.0.0:0` never fails.